addons/hr_recruitment_survey/models/survey.py

```diff
@@ -0,0 +1,19 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+class SurveySurvey(models.Model):
+    _inherit = 'survey.survey'
+
+    survey_type = fields.Selection(selection_add=[('recruitment', 'Recruitment')], ondelete={'recruitment': 'set default'})
+    job_ids = fields.One2many("hr.job", "survey_id")
+
+    @api.depends('survey_type')
+    @api.depends_context('uid')
+    def _compute_allowed_survey_types(self):
```

Comment:
Below that, you can add (and review) this to stay in recruitment if not survey manager, (and perhaps do the same in appraisal if applicable).
```py
    def get_formview_action(self, access_uid=None):
        res = super().get_formview_action(access_uid=access_uid)
        if (self.survey_type == 'recruitment'
                and not self.env.user.has_group('survey.group_survey_manager')
                and self.env.user.has_group('hr_recruitment.group_hr_recruitment_interviewer')):
            if view_id := self.env.ref('hr_recruitment_survey.survey_survey_view_form'):
                res.update({'views': [(view_id.id, 'form')]})
        return res
```

Note also not to forget changing group for `survey_id` in `hr_job_survey_inherit`.